pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -606,6 +607,10 @@ private void commitSegmentMetadataInternal(String realtimeTableName,
 
     // Trigger the metadata event notifier
     _metadataEventNotifierFactory.create().notifyOnSegmentFlush(tableConfig);
+
+    if (StringUtils.isBlank(committingSegmentDescriptor.getSegmentLocation())) {
+      _controllerMetrics.addMeteredTableValue(realtimeTableName, ControllerMeter.SEGMENT_MISSING_DEEP_STORE_LINK, 1);
+    }
```

Comment:
@ankitsultana will this not be equal to the failure metric which we are emitting from `SegmentUploader` classes? Can there be a scenario where these 2 differ? If not, do we need this?

Reply:
It can be different. In Split commit servers upload the file to a temp location, and the file is atomically moved to the final location in the controller. It could be that different replicas try to upload the segment to the deep-store.

Reply:
Hmm makes sense!